src/MusicTheory/Internal/Pitch.elm

```diff
@@ -0,0 +1,151 @@
+module MusicTheory.Internal.Pitch exposing
+    ( Pitch
+    , TransposeError(..)
+    , all
+    , areEnharmonicEqual
+    , doubleFlat
+    , doubleSharp
+    , errorToString
+    , flat
+    , fromPitchClass
+    , natural
+    , octave
+    , pitch
+    , pitchClass
+    , semitones
+    , sharp
+    , transposeDown
+    , transposeUp
+    , tripleFlat
+    , tripleSharp
+    )
+
+import MusicTheory.Internal.PitchClass as PitchClass exposing (Offset, PitchClass)
+import MusicTheory.Interval as Interval exposing (Interval)
+import MusicTheory.Letter as Letter exposing (Letter)
+import MusicTheory.Octave as Octave exposing (Octave, OctaveError(..))
+
+
+type Pitch
+    = Pitch PitchClass Octave
+
+
+type TransposeError
+    = InvalidOctave OctaveError
+    | InternalError
+
+
+pitch : Letter -> Offset -> Octave -> Pitch
+pitch l os o =
+    fromPitchClass o (PitchClass.pitchClass l os)
+
+
+tripleFlat : Offset
+tripleFlat =
+    PitchClass.tripleFlat
+
+
+doubleFlat : Offset
+doubleFlat =
+    PitchClass.doubleFlat
+
+
+flat : Offset
+flat =
+    PitchClass.flat
+
+
+natural : Offset
+natural =
+    PitchClass.natural
+
+
+sharp : Offset
+sharp =
+    PitchClass.sharp
+
+
+doubleSharp : Offset
+doubleSharp =
+    PitchClass.doubleSharp
+
+
+tripleSharp : Offset
+tripleSharp =
+    PitchClass.tripleSharp
+
+
+pitchClass : Pitch -> PitchClass
+pitchClass (Pitch pc _) =
+    pc
+
+
+octave : Pitch -> Octave
+octave (Pitch _ o) =
+    o
+
+
+fromPitchClass : Octave -> PitchClass -> Pitch
+fromPitchClass o p =
+    Pitch p o
+
+
+semitones : Pitch -> Int
+semitones (Pitch pc o) =
+    Octave.semitones o + PitchClass.semitonesNotOctaveBound pc
+
+
+areEnharmonicEqual : Pitch -> Pitch -> Bool
```

Comment:
if we're standardizing on "enharmonic equivalent", then I think this should be renamed to `areEnharmonicEquivalents`